client/lib/domains/index.js

```diff
@@ -144,10 +145,31 @@ function hasMappedDomain( domains ) {
 	return getMappedDomains( domains ).length > 0;
 }
 
+/**
+ * Parse the tld from a given domain name, semi-naively. The function
+ * first parses against a list of tlds that have been sold on WP.com
+ * and falls back to a simplistic "everything after the last dot" approach
+ * if the whitelist failed. This is ultimately not comprehensive as that
+ * is a poor base assumption (lots of second level tlds, etc). However,
+ * for our purposes, the approach should be "good enough" for a long time.
+ *
+ * @param {string}     domainName     The domain name parse the tld from
+ * @return {string}                   The TLD or an empty string
+ */
 function getTld( domainName ) {
 	const lastIndexOfDot = domainName.lastIndexOf( '.' );
 
-	return lastIndexOfDot !== -1 && domainName.substring( lastIndexOfDot + 1 );
+	if ( lastIndexOfDot === -1 ) {
+		return '';
+	}
+
+	let tld = parseDomainAgainstTldList( domainName, getWpcomTldsCachedList() );
```

Comment:
It might be a little clearer if we tuck the list of tlds away inside the function and make the exported function a bit more descriptive of the end result, like `parseDomainAgainstWpcomTlds( domainName )` or `findSupportedTld( domainName )`.